### DISCUSSION TITLE
What is the correct naming convention for what we are currently calling `DistortionModel`?

### DISCUSSION THREAD

**Discussion by bwpriest** [LLNL/MuyGPyS#123] [Ideas]
We currently have `IsotropicDistortion` and `NullDistortion`, and will have `AnisotropicDistortion` in the near future. I initially chose "Distortion" because I believed that we would include the hierarchical nonstationary model in this abstraction. However, as discussed in issue #71, we believe that the hierarchical abstraction will fit into the `Hyperparameter` abstraction. So "Distortion" might not be the right naming convention anymore. 

`IsotropicDistortion` and `AnisotropicDistortion` are functors that take `(..., nn_count, feature_count)` difference tensors and collapse them into `(..., nn_count)` distance tensors. Any future functors filling this role will produce similar `(..., nn_count)` distance tensors, although the input tensors might not have the `(..., nn_count, feature_count)` shape (e.g. graph distances) or might not use difference tensors (e.g. cosine similarity will use `(..., nn_count, feature_count)` elementwise multiplication tensors). We should choose a name that makes sense for our current applications but will also make sense for future applications that we will consider. At present, the distortion model is specified by the user using the `metric` kwarg in their constructor, e.g.
```
matern_fn = Matern(
    nu=Hyperparameter(0.4, (0.1, 1.5)),
    metric=IsotropicDistortion(
        "l2", 
        length_scale=Hyperparameter(1.0),
    ),
)
model = MuyGPS(
    matern_fn, 
    eps=HomoscedasticNoise(),
)
```

===== Alternatives =====

I have a few ideas for other names:
1. `DistanceModel`. I think this most closely matches the intuitions and informal description of the current use case. 
2. `MetricModel`. This conforms with the current `metric` kwarg used to specify the model. Although "metric" and "distance" are used interchangeably in the literature, it is a little more natural to use "metric" when talking about non-euclidean relations, like graph metrics. So I think that `MetricModel` might be more future-proof.

Something that neither of these options addresses are similarity models that are non-metric (e.g. KL divergence doesn't respect the triangle inequality). I'm not sure if there are many/any kernels based upon such similarities, or if we will ever want to support them. If we want to leave that door open, though, we should use language that doesn't exclude them, e.g.
3. `SimilarityModel`. I don't really like this because it is not as intuitive as 1. or 2. However, it is inclusive of non-metric similarities. I'm open to suggestions here.

===== A note on `NullDistortion` =====

`NullDistortion` currently does nothing directly, but is used by `MuyGPyS._test.gp.BenchmarkGP` to create kernel matrices using the assumption that the collapse to distance tensors has already occurred (using `scikit-learn`). This has been a sharp edge for updating tests. I believe that we should ultimately replace it with the usual distortion models, drop the `scikit-learn` dependence, and update those models to ensure that they also work for the dense `(..., batch_count, feature_count) -> (..., batch_count)` use case. We will need to create additional `MuyGPyS.gp.tensors`-style functions, intended only for testing, to generate the dense tensors in question.

**Comment by alecmdunton** [comment]
My vote is for `DistanceModel`. It isn't as restrictive of a nomenclature as `MetricModel`, which I think is too formal in the long run.
